runtime/ms-rest/lib/webResource.js

```diff
@@ -163,7 +163,7 @@ class WebResource {
       }
       let baseUrl = options.baseUrl;
       let url = baseUrl + (baseUrl.endsWith('/') ? '' : '/') + (options.pathTemplate.startsWith('/') ? options.pathTemplate.slice(1) : options.pathTemplate);
-      let segments = url.match(/({\w*\s*\w*})/ig);
+      let segments = url.match(/({\W*\S*\W*})/ig);
```

Comment:
@balajikris @veronicagg Yes those are valid questions. I am also not sure why the initial regex was allowing `white space` / `tabs` between `{}`. 

Secondly, after looking around I saw that technically [swagger path templating](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/2.0.md#path-templating) & [rfc6570](https://tools.ietf.org/html/rfc6570#section-2.3) does not say that `-` or `space` is not allowed. Moreover, [sway](https://github.com/apigee-127/swagger-tools/issues/289) and it's [regex matcher](https://github.com/pillarjs/path-to-regexp/issues/63) intentionally decided not to support `-`. 

The question remains for us is, path [/storage/{storage-account-name}](https://github.com/Azure/azure-rest-api-specs/blob/master/keyvault/2016-10-01/swagger/keyvault.json#L2390) seems to be valid in `keyvault/2016-10-01/swagger/keyvault.json` with parameter defined as ["name": "storage-account-name"](https://github.com/Azure/azure-rest-api-specs/blob/master/keyvault/2016-10-01/swagger/keyvault.json#L2399). The swagger validation tool does not complain about semantic validation but 
1. From linter perspective, do we have any reason for this to be camelCased? For example similar to Rule [M3016](https://github.com/Azure/autorest/blob/master/docs/developer/validation-rules/camel-casing-validation-rules.md)
 2. From model validation perspective how to approach this?

Let me know your thoughts!!